compiler/fir/fir-serialization/src/org/jetbrains/kotlin/fir/serialization/FirElementSerializer.kt

```diff
@@ -607,9 +607,11 @@ class FirElementSerializer private constructor(
 
     private fun transformSuspendFunctionToRuntimeFunctionType(type: ConeClassLikeType): ConeClassLikeType {
         val suspendClassId = type.classId!!
-        val runtimeClassId = FunctionClassDescriptor.Kind.Function.let {
-            ClassId(it.packageFqName, Name.identifier(suspendClassId.relativeClassName.asString().drop("Suspend".length)))
-        }
+        val relativeClassName = suspendClassId.relativeClassName.asString()
+        val kind =
+            if (relativeClassName.startsWith("Suspend")) FunctionClassDescriptor.Kind.Function
```

Comment:
Very minor optimizations: You could just check if it starts with `'K'` to determine if it's a `KSuspendFunction`. And use `replaceFirst` instead of `replace` below.

Reply:
Thanks for the suggestions, and good to learn `replaceFirst`. Both are applied.